app/models/article.rb

```diff
@@ -71,24 +71,7 @@ def content_rendered include_media: true
   end
 
   def related
-    related_articles = {}
-
-    if categories.present?
-      categories.each do |category|
-        articles = []
-        category.articles.english.published[0..7].each do |article|
-          next unless article != self &&
-                      articles.length < 3 &&
-                      !related_articles.values.flatten.include?(article)
-
-          articles << article
-        end
-
-        related_articles[category] = articles
-      end
-    end
-
-    related_articles
+    @related ||= find_related_articles
```

Comment:
Does rails keep the models in memory between request? What I mean here is, do we call `related` twice on the same object in a single request?

I'm not against this change, its more a knowledge question.

Reply:
Activerecord does do query caching, and we also do view caching, so this will only really affect un-cached articles

We do call `#related` more than once per request in the view